README.rst

```diff
@@ -0,0 +1,653 @@
+    export CC=/usr/lib/ccache/gcc
+    export CXX=/usr/lib/ccache/g++
+
+-  Create a ROS 2 workspace for ASAN related tasks (do not compile yet!)
+
+.. code:: bash
+
+    cd ~/ros2_asan_ws
+    wget https://raw.githubusercontent.com/ros2/ros2/master/ros2.repos
+    vcs-import src < ros2.repos
+
+Skipping unnecessary packages (Recommended)
+---------------------
+
+Unless you are specifically testing these packages, you may ignore them to save time:
+
+.. code:: bash
+
+    touch src/ros2/common_interfaces/actionlib_msgs/COLCON_IGNORE
+    touch src/ros2/common_interfaces/common_interfaces/COLCON_IGNORE
+    touch src/ros2/rosidl_typesupport_opensplice/opensplice_cmake_module/COLCON_IGNORE
+    touch src/ros2/rmw_fastrtps/rmw_fastrtps_dynamic_cpp/COLCON_IGNORE
+    touch src/ros2/rmw_opensplice/rmw_opensplice_cpp/COLCON_IGNORE
+    touch src/ros2/ros1_bridge/COLCON_IGNORE
+    touch src/ros2/rosidl_typesupport_opensplice/rosidl_typesupport_opensplice_c/COLCON_IGNORE
+    touch src/ros2/rosidl_typesupport_opensplice/rosidl_typesupport_opensplice_cpp/COLCON_IGNORE
+    touch src/ros2/common_interfaces/shape_msgs/COLCON_IGNORE
+    touch src/ros2/common_interfaces/stereo_msgs/COLCON_IGNORE
+    touch src/ros2/common_interfaces/trajectory_msgs/COLCON_IGNORE
+
+Compiling the code and running the tests
+----------------------------------------
+
+-  First, compile all packages:
+
+.. code:: bash
+
+    cd ~/ros2_asan_ws  # you will need to be exactly in this directory
+    # Those flags are similar to what the nightly job uses.
+    colcon build --build-base=build-asan --install-base=install-asan \
+        --cmake-args -DOSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS=ON \
+                     -DINSTALL_EXAMPLES=OFF -DSECURITY=ON --no-warn-unused-cli \
+                     -DCMAKE_BUILD_TYPE=Debug \
+        --mixin asan-gcc \
+        --packages-up-to test_communication \
+        --symlink-install
+
+.. code:: bash
+
+    # Equivalent command for tsan:
+    # You can either use different workspaces or the same one.
+    colcon build --build-base=build-tsan --install-base=install-tsan \
+        --cmake-args -DOSRF_TESTING_TOOLS_CPP_DISABLE_MEMORY_TOOLS=ON \
+                     -DINSTALL_EXAMPLES=OFF -DSECURITY=ON --no-warn-unused-cli \
+                     -DCMAKE_BUILD_TYPE=Debug \
+        --mixin tsan \
+        --packages-up-to test_communication \
+        --symlink-install
+
+⚠️ **IMPORTANT:** Do not forget to pass ``-DCMAKE_BUILD_TYPE=Debug`` to
+ensure that debugging symbols are generated. This allows ASAN/MSAN to
+report files and line numbers in backtraces.
+
+Once the compilation is finished, colcon will report that all packages
+have been compiled but some had “stderr output”. This is fine.
+
+-  To run the tests for ASan:
+
+.. code:: bash
+
+    cd ~/ros2_asan_ws  # you will need to be exactly in this directory
+    colcon test --build-base=build-asan --install-base=install-asan \
+        --packages-up-to test_communication
```

Comment:
perhaps adding `--event-handler sanitizer-report+` ?

Reply:
What is the reason to disable the event handler by default? Is there a significant overhead associated to it? It would also only need to do anything if the extension is installed and the build uses a sanitizer.

Reply:
Agreed that it'd make sense to enable the plugin if it's installed and we `colcon build` with a sanitizer. Is that information available during `colcon test`? If we were to enable it always, we'd output a blank CSV report after running `colcon test` and the code was not built with a sanitizer.

I haven't measured, but I don't think this plugin adds significant overhead. Memory-wise, it holds onto text for a few sample stack traces (we saw 40ish in our first tests). CPU-wise, there's the cost of doing 1-4 regex searches on every line of stdout/stderr output.

Reply:
> If we were to enable it always, we'd output a blank CSV report after running `colcon test`

You could easily update the logic to not write the csv file in that case ;-)

Reply:
Added https://github.com/colcon/colcon-sanitizer-reports/issues/13